src/Service/TraceContextService.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+declare(strict_types=1);
+
+namespace DR\SymfonyRequestId\Service;
+
+use DR\SymfonyRequestId\TraceContext;
+use InvalidArgumentException;
+
+/**
+ * @internal
+ */
+class TraceContextService
+{
+    public const HEADER_TRACEPARENT = 'traceparent';
+    public const HEADER_TRACESTATE  = 'tracestate';
+
+    public function validateTraceParent(string $traceParent): bool
+    {
+        return preg_match('/^[0-9a-f]{2}-[0-9a-f]{32}-[0-9a-f]{16}-[0-9a-f]{2}$/i', $traceParent) === 1;
+    }
+
+    public function renderTraceParent(TraceContext $traceContext): string
+    {
+        return sprintf(
+            '%s-%s-%s-%s',
+            $traceContext->getVersion(),
+            $traceContext->getTraceId(),
+            $traceContext->getTransactionId(),
+            $traceContext->getFlags()
+        );
+    }
+
+    public function renderTraceState(TraceContext $traceContext): string
+    {
+        $traceState = [];
+        foreach ($traceContext->getTraceState() as $key => $value) {
+            $traceState[] = sprintf('%s=%s', $key, $value);
+        }
+
+        return implode(',', $traceState);
+    }
+
+    public function parseTraceContext(string $traceParent, string $traceState): TraceContext
```

Comment:
Move this method below the supports method and pass the Request object.

Pref: `public function getTraceContext(Request $request): TraceContext`